serializers/avro/src/main/java/io/pravega/schemaregistry/serializer/avro/impl/AvroSerializer.java

```diff
@@ -27,30 +26,32 @@
 import java.io.OutputStream;
 
 public class AvroSerializer<T> extends AbstractSerializer<T> {
-    private final AvroSchema<T> avroSchema;
+    private final org.apache.avro.Schema avroSchema;
+    private final SpecificDatumWriter<T> specificDatumWriter;
+    private final GenericDatumWriter<T> genericDatumWriter;
+    private final ReflectDatumWriter<T> reflectDatumWriter;
+
     public AvroSerializer(String groupId, SchemaRegistryClient client, AvroSchema<T> schema,
                    Encoder encoder, boolean registerSchema) {
         super(groupId, client, schema, encoder, registerSchema, true);
-        this.avroSchema = schema;
+        this.avroSchema = schema.getSchema();
+        this.specificDatumWriter = new SpecificDatumWriter<>(avroSchema);
```

Comment:
do we want to instantiate all three of the writer types or is there a way to figure out which one is relevant based on the type T?

Reply:
Looking into profiled AvroSerializer.serialize method, we can see there is no impact of checking type
```java
 if (IndexedRecord.class.isAssignableFrom(var.getClass())) {
            if (SpecificRecord.class.isAssignableFrom(var.getClass())) {
```
See
![image](https://user-images.githubusercontent.com/858788/95058309-0201d100-0700-11eb-8bc3-6f21ffd74295.png)

Reply:
that looks good.